helpers.js

```diff
@@ -1,141 +1,77 @@
-/* ----------- HTML Parsing functions ----------- */
-
-// Returns elements with a certain attribute
-function getElementsByAttribute(element, attribute, classToFind) {
-  var data = [];
-  var descendants = element.getDescendants();
-  descendants.push(element);
-  for(i in descendants) {
-    var elt = descendants[i].asElement();
-    if(elt != null) {
-      var classes = elt.getAttribute(attribute);
-      if(classes != null) {
-        classes = classes.getValue();
-        if(classes == classToFind) data.push(elt);
-        else {
-          classes = classes.split(' ');
-          for(j in classes) {
-            if(classes[j] == classToFind) {
-              data.push(elt);
-              break;
-            }
-          }
-        }
-      }
-    }
-  }
-  return data;
-}
-
-// Returns elements with a certain class
-function getElementsByClassName(element, classToFind) {
-  var data = [];
-  var descendants = element.getDescendants();
-  descendants.push(element);
-  for(i in descendants) {
-    var elt = descendants[i].asElement();
-    if(elt != null) {
-      var classes = elt.getAttribute('class');
-      if(classes != null) {
-        classes = classes.getValue();
-        if(classes == classToFind) data.push(elt);
-        else {
-          classes = classes.split(' ');
-          for(j in classes) {
-            if(classes[j] == classToFind) {
-              data.push(elt);
-              break;
-            }
-          }
-        }
-      }
-    }
-  }
-  return data;
-}
-
-// Returns elements with a certain tag name
-function getElementsByTagName(element, tagName) {
-  var data = [];
-  var descendants = element.getDescendants();
-  for(i in descendants) {
-    var elt = descendants[i].asElement();
-    if( elt !=null && elt.getName()== tagName) data.push(elt);
-  }
-  return data;
-}
-
 /* ----------- Data processing functions ----------- */
 function isISIN(id) {
   // ISIN have 12 characters
   // Not the most robust way (TODO: improve it)
   return id.length == 12;
 }
 
+function stripCharacters_(text) {
+  return text.trim().replace(/\n/g, '').replace(/\t/g, '');
+}
+
 function processNav(nav) {
+  nav = stripCharacters_(nav);
```

Comment:
Would it make sense to strip these characters just once in `fetchURL`, just before caching the result?

Reply:
Thing is this is stripping e.g. whitespace at the start and end of lines as well - that can't be done in `fetchURL`. I think it's good to have a sanitising function here as well.